src/detect-bytejump.h

```diff
@@ -40,9 +40,11 @@
 #define DETECT_BYTEJUMP_DCE       BIT_U16(6) /**< "dce" enabled */
 #define DETECT_BYTEJUMP_OFFSET_BE BIT_U16(7) /**< "byte extract" enabled */
 #define DETECT_BYTEJUMP_END       BIT_U16(8) /**< "from_end" jump */
+#define DETECT_BYTEJUMP_NBYTES_VAR BIT_U16(9) /**< nbytes string*/
 
 typedef struct DetectBytejumpData_ {
     uint8_t nbytes;                   /**< Number of bytes to compare */
+    uint8_t nbytes_idx;               /**< If nbytes is a string, idx of var */
```

Comment:
Not sure if we can easily avoid it, but this grows the struct.

Old:

```c
struct DetectBytejumpData_ {
	uint8_t                    nbytes;               /*     0     1 */
	uint8_t                    base;                 /*     1     1 */
	uint16_t                   flags;                /*     2     2 */
	uint32_t                   multiplier;           /*     4     4 */
	int32_t                    offset;               /*     8     4 */
	int32_t                    post_offset;          /*    12     4 */

	/* size: 16, cachelines: 1, members: 6 */
	/* last cacheline: 16 bytes */
};
```

New:
```c
struct DetectBytejumpData_ {
	uint8_t                    nbytes;               /*     0     1 */
	uint8_t                    nbytes_idx;           /*     1     1 */
	uint8_t                    base;                 /*     2     1 */

	/* XXX 1 byte hole, try to pack */

	uint16_t                   flags;                /*     4     2 */

	/* XXX 2 bytes hole, try to pack */

	uint32_t                   multiplier;           /*     8     4 */
	int32_t                    offset;               /*    12     4 */
	int32_t                    post_offset;          /*    16     4 */

	/* size: 20, cachelines: 1, members: 7 */
	/* sum members: 17, holes: 2, sum holes: 3 */
	/* last cacheline: 20 bytes */
};
```

Reply:
There's a few things we could try:
- `nbytes` can't be more than `23` so perhaps it coult be a 5-bit value and the index 3-bits?
- `flags` could contain a check if `nbytes` is an index or not
- Perhaps `multiplier` could be smaller?

Thoughts @victorjulien ?

Reply:
I removed `nbytes_idx` and decreased the size of `multiplier` since it's value range is [1, 65535]